main.go

```diff
@@ -0,0 +1,357 @@
+		os.Exit(1)
+	}
+	if csctlConfig.Config.Provider.Type != provider {
+		fmt.Printf("Wrong provider in %s. Expected %s\n", clusterStackPath, provider)
+		os.Exit(1)
+	}
+	releaseDir := os.Args[3]
+	_, err = os.Stat(releaseDir)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	registryConfigPath := os.Args[4]
+
+	method := csctlConfig.Config.Provider.Config["method"]
+	switch method {
+	case "get":
+		// Copy config.yaml to releaseDir as node-images.yaml
+		dest := filepath.Join(releaseDir, "node-images.yaml")
+		if err := copyFile(configFilePath, dest); err != nil {
+			fmt.Printf("Error copying config.yaml to releaseDir: %v\n", err)
+			os.Exit(1)
+		}
+		fmt.Println("config.yaml copied to releaseDir as node-images.yaml successfully!")
+	case "build":
+		for _, image := range config.OpenStackNodeImages {
+			if image.ImageDir == "" {
+				fmt.Printf("No images to build, image directory is not defined in config.yaml file")
+				os.Exit(1)
+			}
+
+			// Construct the path to the image folder
+			packerImagePath := filepath.Join(clusterStackPath, "node-images", image.ImageDir)
+
+			if _, err := os.Stat(packerImagePath); err != nil {
+				fmt.Printf("Image folder %s does not exist\n", packerImagePath)
+				os.Exit(1)
+			}
+			fmt.Println("Running packer build...")
+			// Warning: variables like build_name and output_directory must exist in packer ariables file like in example
+			// #nosec G204
+			cmd := exec.Command("packer", "build", "-var", "build_name="+image.ImageDir, "-var", "output_directory="+outputDirectory, packerImagePath)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			if err := cmd.Run(); err != nil {
+				fmt.Printf("Error running packer build: %v\n", err)
+				os.Exit(1)
+			}
+			fmt.Println("Packer build completed successfully.")
+
+			// Todo: Use --node-image-registry flag to pass path to registry.yaml
+			_, err = os.Stat(registryConfigPath)
+			if err != nil {
+				fmt.Println(err.Error())
+				os.Exit(1)
+			}
+			// Get the current working directory
+			currentDir, err := os.Getwd()
+			if err != nil {
+				fmt.Printf("Error getting current working directory: %v\n", err)
+				os.Exit(1)
+			}
+
+			// Path to the image created by the packer
+			// Warning: name of the image created by packer should have same name as the name of the image folder in node-images
+			ouputImagePath := filepath.Join(currentDir, outputDirectory, image.ImageDir)
+
+			// Push the built image to S3
+			if err := pushToS3(ouputImagePath, image.ImageDir, registryConfigPath); err != nil {
+				fmt.Printf("Error pushing image to S3: %v\n", err)
+				os.Exit(1)
+			}
+
+			// Update URL in config.yaml if it is necessary
+			if err := updateURLNodeImages(configFilePath, registryConfigPath, image.ImageDir); err != nil {
+				fmt.Printf("Error updating URL in config.yaml: %v\n", err)
+				os.Exit(1)
+			}
+			// Copy config.yaml to releaseDir as node-images.yaml
+			dest := filepath.Join(releaseDir, "node-images.yaml")
```

Comment:
Am I missing something or is this the same for all images in the for loop?

Reply:
see https://github.com/SovereignCloudStack/csctl-plugin-openstack/pull/4/commits/df095ac51d2ced4758a0d4817db255e0ff30a003